src/combobox/ComboBox.ts

```diff
@@ -0,0 +1,360 @@
+			openOnFocus,
+			onFocus,
+			onRequestResults
+		} = this.properties;
+
+		const value = (<HTMLInputElement> event.target).value;
+
+		this._focused = true;
+		this._ignoreBlur = false;
+		onFocus && onFocus(value);
+
+		if (openOnFocus && !this._ignoreFocus) {
+			this._open = true;
+			onRequestResults && onRequestResults(value);
+		}
+
+		this._ignoreFocus = false;
+	}
+
+	onInputBlur(event: FocusEvent) {
+		const { onBlur } = this.properties;
+
+		if (this._ignoreBlur) {
+			return;
+		}
+
+		this.updateSelectedIndex(Operation.reset);
+		this._open = false;
+		this._focused = false;
+		onBlur && onBlur((<HTMLInputElement> event.target).value);
+		this.invalidate();
+	}
+
+	onInputKeyDown(event: KeyboardEvent) {
+		this._handlers[event.keyCode] && this._handlers[event.keyCode].call(this, event);
+	}
+
+	onArrowClick() {
+		const {
+			onRequestResults,
+			inputProperties
+		} = this.properties;
+
+		if (inputProperties && (inputProperties.disabled || inputProperties.readOnly)) {
+			return;
+		}
+
+		this._open = true;
+		this._focused = true;
+		this._inputElement && onRequestResults && onRequestResults(this._inputElement.value);
+	}
+
+	onClearClick() {
+		const { onChange } = this.properties;
+		this._focused = true;
+		onChange && onChange('');
+	}
+
+	onResultMouseEnter(index: number) {
+		this.updateSelectedIndex(undefined, index);
+		this.invalidate();
+	}
+
+	onResultMouseDown() {
+		this._ignoreBlur = true;
+	}
+
+	onResultMouseUp() {
+		const { results } = this.properties;
+
+		if (!results || this._selectedIndex === undefined) {
+			return;
+		}
+
+		this.selectResult(results[this._selectedIndex]);
+	}
+
+	selectResult(result: any) {
+		const {
+			autoBlur,
+			onChange
+		} = this.properties;
+
+		this.updateSelectedIndex(Operation.reset);
+		this._open = false;
+		this._focused = !autoBlur;
+		this._ignoreFocus = true;
+		onChange && onChange(this.getResultLabel(result));
+	}
+
+	getResultLabel(result: any): string {
+		const { getResultLabel } = this.properties;
+		return getResultLabel ? getResultLabel(result) : result;
+	}
+
+	renderMenu(results: any[]) {
+		const menu = w('result-menu', <ResultMenuProperties> {
+			registry: this.registry,
+			results: results,
+			selectedIndex: this._selectedIndex,
+			getResultLabel: this.getResultLabel,
+			skipResult: () => {
+				this.updateSelectedIndex(this._direction);
+			},
+			onResultMouseEnter: this.onResultMouseEnter,
+			onResultMouseDown: this.onResultMouseDown,
+			onResultMouseUp: this.onResultMouseUp
+		});
+
+		return results.length > 0 ? menu : null;
+	}
+
+	render(): DNode {
+		const {
+			clearable,
+			results = [],
+			inputProperties = {},
+			value = '',
+			onMenuChange
+		} = this.properties;
+
+		const menu = this.renderMenu(results);
+
+		menu && this._open && !this._wasOpen && onMenuChange && onMenuChange(true);
+		menu && !this._open && this._wasOpen && onMenuChange && onMenuChange(false);
+
+		this._wasOpen = this._open;
+
+		return v('div', {
```

Comment:
This is nitpicky, especially given that the `display` is overridden in the CSS, but since native selects are inline elements, I wonder whether this should use an inline-by-default element as well.